### PR TITLE
Miguelhar.plat 93.implement deploy id restrictions

### PR DESCRIPTION
**Pull request by miguelhar**

### CODE REVIEW

File: .github/workflows/terraform-checks.yml [hunk 1]
```diff
@@ -67,21 +67,22 @@ jobs:
...
+  #       uses: actions/checkout@master
+  #     - uses: actions/setup-python@v3
+  #     - name: Install terrascan
+  #       run: >-
+  #         curl -sL "$(curl -sL https://api.github.com/repos/accurics/terrascan/releases/latest | grep -o -E "https://.+?_Linux_x86_64.tar.gz")" > terrascan.tar.gz &&
+  #         tar -xf terrascan.tar.gz terrascan && rm terrascan.tar.gz && install terrascan /usr/local/bin && rm terrascan
+  #     - name: Run Terrascan
+  #       uses: pre-commit/action@v3.0.0
+  #       with:
+  #         extra_args: terrascan --all-files
```
Comment (fraenkel) [inline .github/workflows/terraform-checks.yml:88]: what are we waiting on?

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
 .terraform
 **.terraform.lock.hcl
 **kubeconfig
+results.sarif
```
Comment (fraenkel) [inline .gitignore:4]: a bit random

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -49,7 +49,9 @@ repos:
...
-      - id: terrascan
-        args:
-          - '--args=--verbose'
-          - '--args=--skip-rules=AC_GCP_0016,AC_GCP_0234,AC_GCP_0006'
+      ## Merge back the terrascan id when the following error is merged:
+      ## Unsupported argument; An argument named "nullable" is not expected here.
+      # - id: terrascan ## Merge back when there is a new version upstream
+      #   args:
+      #     - '--args=--verbose'
+      #     - '--args=--skip-rules=AC_GCP_0016,AC_GCP_0234,AC_GCP_0006'
```
Comment (fraenkel) [inline .pre-commit-config.yaml:57]: is there a PR or Issue to track?

File: .github/workflows/terraform-checks.yml (hunk 1 above)
Comment (miguelhar) [reply .github/workflows/terraform-checks.yml:88]: some 1.2 features are not supported yet, causing the scan to fail

File: .gitignore (hunk 2 above)
Comment (miguelhar) [reply .gitignore:4]: produced by the precommit scan

File: .pre-commit-config.yaml (hunk 3 above)
Comment (miguelhar) [reply .pre-commit-config.yaml:57]: https://github.com/tenable/terrascan/issues/1331

File: .github/workflows/terraform-checks.yml (hunk 1 above)
Comment (fraenkel) [reply .github/workflows/terraform-checks.yml:88]: do we have an issue to track re-enabling?

**Review by fraenkel — APPROVED** [review]
would create the issue to track what is needed to re-enable.

**Pull request merged**